### PR TITLE
chore: [SIW-1118] Added new workspaces apps/io-wallet-web-func and packages/io-wallet-common

### PR DESCRIPTION
**Pull request by silvicir**
#### List of Changes

1. Added new workspace `apps/io-wallet-web-func`. This is the function app that includes the wallet API called by io-web. The health check endpoint is already included
2. Added `asyncapi.yaml` to the `io-wallet-user-func` function app
3. Added `packages/io-wallet-common` workspace with code shared between the two functions. Already moved the shared code into the new workspace
4. Some lint fixes
5. Updated `turbo.json` file since the file does not pass the new turbo configuration
6. Updated node version to `20.13.1` (the same as azure)

#### Motivation and Context

An additional function app is needed for the API called by io-web. Until now, there was only one workspace. With the addition of another workspace in the monorepo, I have added a shared code package between the two

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-wallet-user-func/asyncapi.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+              type: object
+              properties:
+                kty:
+                  type: string
+                x:
+                  type: string
+                y:
+                  type: string
+                crv:
+                  type: string
```
Comment (grausof) [inline apps/io-wallet-user-func/asyncapi.yaml:38]: Sometimes hardwareKey may contain optional attributes such as KID etc. What do you think about encoding the entire key in base64?

File: apps/io-wallet-user-func/tsup.config.js [hunk 2]
```diff
@@ -6,5 +6,5 @@ export default defineConfig({
   sourcemap: true,
   clean: true,
   minify: false,
-  target: "node18",
+  target: "node20",
```
Comment (grausof) [inline apps/io-wallet-user-func/tsup.config.js:9]: I think it should also be updated [.nvmrc ](https://github.com/pagopa/io-wallet/blob/master/.nvmrc)

File: apps/io-wallet-web-func/asyncapi.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+              type: object
+              properties:
+                kty:
+                  type: string
+                x:
+                  type: string
+                y:
+                  type: string
+                crv:
+                  type: string
```
Comment (grausof) [inline apps/io-wallet-web-func/asyncapi.yaml:28]: ditto

**Review by grausof — APPROVED** [review]

File: apps/io-wallet-user-func/tsup.config.js (hunk 2 above)
Comment (silvicir) [reply apps/io-wallet-user-func/tsup.config.js:9]: do we want to delete the file? do we use nvm?

Comment (grausof) [reply apps/io-wallet-user-func/tsup.config.js:9]: I think `nodenv` should be used

File: apps/io-wallet-user-func/asyncapi.yaml (hunk 1 above)
Comment (grausof) [reply apps/io-wallet-user-func/asyncapi.yaml:38]: OK for now. Then we settle in another PR

**Pull request merged**